vertx-web-client/src/main/java/io/vertx/ext/web/client/PathTemplate.java

```diff
@@ -0,0 +1,41 @@
+package io.vertx.ext.web.client;
+
+import io.vertx.codegen.annotations.VertxGen;
+import io.vertx.ext.web.client.impl.PathTemplateImpl;
+
+/**
+ * Path templates for Web Client. Supports the same syntax of Vert.x Web path templates:
+ * You can define parameters placeholders with {@code :} followed by the parameter name. For instance:<br/>
+ * <ul>
+ *     <li>{@code "/:name/:id?hello=name"} with parameters (name: francesco),(id: 10) will translate to {@code /francesco/10?hello=name}</li>
+ *     <li>{@code "/api/v=:api_version/users/:id"} with parameters (api_version: 1),(id: 10) will translate to {@code /api/v=1/users/10}</li>
+ * </ul>
+ *
+ * This object can be reused for multiple requests
+ *
+ * @author Francesco Guardiani @slinkydeveloper
+ */
+@VertxGen
+public interface PathTemplate {
```

Comment:
I am thinking why not just use a JsonObject instead? where all values are coalesced to String, plus it would also allow the usage of arrays, so for example you could have:

```
{ key: [1, 2, 3] }

would become:

key=1&key=2&key=3
```

When the key to replace happens, after the query symbol `?`

Reply:
Ok, replying to myself, Json would be to abstract as it can't differentiate from escaped and non escaped, but perhaps this class could allow arrays too

Reply:
The initial idea was to have a path templating language same as vertx-web one.
Perhaps to support something more complex we need to "define" some rules/restrictions. E.g. how an array is expanded in a query param? As `key=1&key=2&key=3`? or as `key=1,2,3`? How is expanded in a path parameter? as `/1/2/3` or as `/1,2,3`?
I would like to propose to serialize to `key=1&key=2&key=3` & `/1/2/3`, wdyt?